### PR TITLE
UCT/IB/MLX5: Refactor MLX5 common code.

### PR DESCRIPTION
**Pull request by yosefe**

### CODE REVIEW

**Comment by mellanox-github** [comment]
Test FAILed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr__NEW/1/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://bgate.mellanox.com/jenkins/job/gh-ucx-pr/2614/ for details.

**Comment by mellanox-github** [comment]
Test PASSed.
See http://hpc-master.lab.mtl.com:8080/job/hpc-ucx-pr/1968/ for details (Mellanox internal link).

**Comment by yosefe** [comment]
@shssf can you pls review?

File: src/uct/ib/mlx5/ib_mlx5.c [hunk 1]
```diff
@@ -234,43 +262,43 @@ ucs_status_t uct_ib_mlx5_get_txwq(uct_worker_h worker, struct ibv_qp *qp,
...
 #endif
-    ucs_assert_always(wq->bb_max > 0);
-    memset(wq->qstart, 0, wq->qend - wq->qstart); 
+    ucs_assert_always(txwq->bb_max > 0);
+    memset(txwq->qstart, 0, txwq->qend - txwq->qstart);
     return UCS_OK;
 } 
 
-void uct_ib_mlx5_put_txwq(uct_worker_h worker, uct_ib_mlx5_txwq_t *wq)
+void uct_ib_mlx5_txwq_cleanup(uct_worker_h worker, uct_ib_mlx5_txwq_t* txwq)
```
Comment (yosefe) [inline src/uct/ib/mlx5/ib_mlx5.c]: these are slow-path functions

**Comment by shssf** [comment]
:+1:

**Pull request merged**